conbench/api/history.py

```diff
@@ -37,8 +26,19 @@ def get(self, benchmark_id):
         tags:
           - History
         """
-        history = self._get(benchmark_id)
-        return self.serializer.many.dump(history)
+        # TODO: think about the case where samples if of zero length. Can this
+        # happen? If it can happen: which response would we want to emit to the
+        # HTTP client? An empty array, or something more convenient?
+        try:
+            samples = get_history_for_benchmark(benchmark_result_id=benchmark_id)
+        except NotFound:
+            self.abort_404_not_found()
+
+        # if performance is a concern then https://pypi.org/project/orjson/
+        # promises to be among the fastest for serializing python dataclass
+        # instances into JSON. Note: wrap this into an array if there is just 1
+        # sample, for consistency (clients can expect an array).
+        return f.jsonify([s._dict_for_api_json() for s in samples])
```

Comment:
Now, the IDE knows what `s` is:
![image](https://user-images.githubusercontent.com/265630/223407820-2614ae45-e42d-4aac-b927-61441153bd70.png)